pkg/services/serviceaccounts/proxy/service_test.go

```diff
@@ -0,0 +1,259 @@
+package proxy
+
+import (
+	"context"
+	"testing"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/apikey"
+	"github.com/grafana/grafana/pkg/services/extsvcauth/extsvcaccounts"
+	"github.com/grafana/grafana/pkg/services/serviceaccounts"
+	"github.com/stretchr/testify/assert"
+)
+
+type FakeServiceAccountsService struct {
```

Comment:
Instead of re-creating this, I'd suggest moving the Fake from: `pkg/services/serviceaccounts/api/api_test.go` to `pkg/services/serviceaccounts/tests` add use it in both API tests and Proxy tests.

Reply:
This is a great idea, but do you think it is within the scope of this pr? I don't mind do this in a follow-up PR just to keep a clear scoped between PRs, rather than moving the goalpost

Reply:
Depends how big of a change it is, but if it's only three files, let's do it once and for all now  :100: